test/e2eprovider/server/server.go

```diff
@@ -0,0 +1,177 @@
+// +build e2e
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+	"sigs.k8s.io/secrets-store-csi-driver/test/e2eprovider/types"
+
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	secrets = map[string]string{
+		"foo": "secret",
+		"fookey": `-----BEGIN PUBLIC KEY-----
+This is mock key
+-----END PUBLIC KEY-----`,
+	}
+
+	podCache = make(map[string]bool)
+
+	podUIDAttribute = "csi.storage.k8s.io/pod.uid"
+)
+
+// Server is a mock csi-provider server
+type Server struct {
+	grpcServer *grpc.Server
+	listener   net.Listener
+	SocketPath string
+	network    string
+}
+
+// NewE2EProviderServer returns a mock csi-provider grpc server
+func NewE2EProviderServer(endpoint string) (*Server, error) {
+	network := ""
+	address := ""
+	if strings.HasPrefix(strings.ToLower(endpoint), "unix://") || strings.HasPrefix(strings.ToLower(endpoint), "tcp://") {
+		s := strings.SplitN(endpoint, "://", 2)
+		if s[1] != "" {
+			network = s[0]
+			address = s[1]
+		} else {
+			return nil, fmt.Errorf("Invalid endpoint: %s", endpoint)
+		}
+	}
+
+	server := grpc.NewServer()
+	s := &Server{
+		grpcServer: server,
+		SocketPath: address,
+		network:    network,
+	}
+
+	v1alpha1.RegisterCSIDriverProviderServer(server, s)
+
+	return s, nil
+}
+
+// Start starts the mock csi-provider server
+func (m *Server) Start() error {
+	var err error
+
+	m.listener, err = net.Listen(m.network, m.SocketPath)
+	if err != nil {
+		return err
+	}
+
+	klog.InfoS("Listening for connections on address:", m.listener.Addr())
```

Comment:
```suggestion
	klog.InfoS("Listening for connections", "address", m.listener.Addr())
```
this is using structured logging

Reply:
done